boot/zephyr/include/mcuboot_config/mcuboot_config.h

```diff
@@ -140,6 +140,16 @@
 #define MCUBOOT_DECOMPRESS_IMAGES
 #endif
 
+/* Invoke hashing functions directly on storage. This requires for device
+ * to be able to map storage to address space or RAM.
+ */
+#ifdef CONFIG_BOOT_IMG_HASH_DIRECTLY_ON_STORAGE
+#ifdef MCUBOOT_ENC_IMAGES
+#error "Direct hash check is currently not supported when encrypted images are enabled"
+#endif
```

Comment:
this check can go really, it's already enforced by Kconfig, doesn't make sense to check for something that isn't possible